### PR TITLE
[sparrow] Add `sparrow`

### PR DESCRIPTION
**Pull request by jjerphan**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by jjerphan** [comment]
Hi @MonicaLiu0311,

I have been trying to understand why one of the builds fails but I really cannot the reason behind it.

[The error message](https://dev.azure.com/vcpkg/public/_build/results?buildId=100628&view=logs&j=878666d5-db33-5b27-9e7d-b0c7ee352005&t=3980505e-92f5-53ef-75a8-4f2e70b97516&l=2426) does not makes a lot of sense to me, especially given that the layout of this PR is identical to other PRs' such as https://github.com/microsoft/vcpkg/pull/37314/.

Could you help?

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
sparrow is header-only and can be used from CMake via:

    find_path(SPARROW_INCLUDE_DIRS "sparrow/buffer.hpp")
    target_include_directories(main PRIVATE ${SPARROW_INCLUDE_DIRS})
```

**Review by MonicaLiu0311 — DISMISSED** [review]

**Comment by jjerphan** [comment]
Thank you, @MonicaLiu0311.

**Comment by BillyONeal** [comment]
> * The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.

This does not seem to be correct. This project is not associated with `sparrow` on repology, and first search results point to https://github.com/ropas/sparrow , not https://github.com/xtensor-stack/sparrow .

Would you accept xtensor-stack-sparrow ?

**Comment by jjerphan** [comment]
`x-sparrow` would work for us as the ownership might be shared beyond the xtensor-stack's organization.

**Comment by dg0yt** [comment]
Hm, `x-` has its own world of associations. Is `xtensor-` a candidate?

**Comment by JohanMabille** [comment]
> Is xtensor- a candidate?

We'd like to keep it for xtensor bindings on sparrow. Would cpp-sparrow be ok (as suggested by the last change)?

**Comment by BillyONeal** [comment]
@ vcpkg team: Are we OK with `cpp-sparrow`?
Also, should we say this depends on https://github.com/microsoft/vcpkg/pull/37457 so that the patch is unnecessary? (I'm a bit uneasy patching a version that upstream explicitly picked)

**Comment by JohanMabille** [comment]
The thing is that `sparrow` will be transfered to the man-group org when it's more stable. Would man-group-sparrow be acceptable?

**Comment by BillyONeal** [comment]
> The thing is that `sparrow` will be transfered to the man-group org when it's more stable. Would man-group-sparrow be acceptable?

Can't we just rename the port if/when that happens?

**Comment by jjerphan** [comment]
Does renaming the port makes the old name available again? If so, could name this port `xtensor-stack-sparrow` for now, and rename it when `xtensor-stack/sparrow` becomes `man-group/sparrow`?

**Comment by BillyONeal** [comment]
>Does renaming the port makes the old name available again?

No, in general people would have to update their references. But they'd have to do that referencing the github repo in the first place.

**Comment by jjerphan** [comment]
I am trying to understand the errors on Windows though I do not have a machine running this OS.

**Comment by jjerphan** [comment]
Hello @BillyONeal,

The project has been transferred to another organization, and this PR is now ready for another review. :slightly_smiling_face:

**Comment by BillyONeal** [comment]
The naming concern remains. Would you accept `man-group-sparrow` to follow the `[github organization]-[repo]` convention?

**Comment by jjerphan** [comment]
I just have renamed the port accordingly.

**Comment by jjerphan** [comment]
Can we actually reconsider `sparrow` as a name since https://github.com/ropas/sparrow is a small, unmaintained OCAML project?

**Comment by SylvainCorlay** [comment]
I would love this to be named sparrow. It would save us the trouble of having to tell people to use a different name on vcpkg.

**Comment by BillyONeal** [comment]
The problem with the name sparrow is again, not in repology, and search hits lead to

* ropas/sparrow
* Sparrow-lang/sparrow

**Comment by BillyONeal** [comment]
@AugP
@ras0219-msft
@data-queue
@JavierMatosD
and I discussed this today.

We affirm what we said back in March, there's too much potential for confusion and we want the name to be `man-group-sparrow`.

**Comment by MonicaLiu0311** [comment]
Ping @jjerphan

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

File: ports/sparrow/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    SHA512 388c7401c9168cb389f619f196a02f69e54934a66d06828d9d3cc5ea64c858f6512ccaca94309eebbeefbc57a9e864f0f203a187413e4d8ce688791e00d39757
+    HEAD_REF main
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS
+        -DBUILD_TESTS=OFF
+        -DUSE_DATE_POLYFILL=ON
+)
```
Comment (Alex-PLACET) [inline ports/sparrow/portfile.cmake:16]: In version 0.4 you can now compile shared library by passing: -DSPARROW_BUILD_SHARED otherwise it's static

**Pull request closed without merging**